src/MTGCore.ClientApp/ClientApp/src/components/DeckList.vue

```diff
@@ -1,30 +1,67 @@
 <template>
-  <div class="form-group row" style="margin-bottom:0px">
-    <label class="col-lg-3">Deck</label>
-    <select class="form-control float-right col-lg-9" @change="onchange" v-model="selectedOption">
-      <option v-for="item in deckList" v-bind:Key="item.id" v-bind:value="item.id">{{item.title}}</option>
-    </select>
-  </div>
+    <div>
+        <div class="form-group row" style="margin-bottom:0px">
+            <div class="input-group">
+                <div class="input-group-prepend float-right">
+                    <input type="button" class="btn btn-default input-group-text" value="Add New Deck" data-toggle="modal" data-target="#TitleModal">
+                </div>
+                <select class="form-control float-right col-lg-9" @change="onchange" v-model="selectedOption">
+                    <option v-for="item in deckList" v-bind:Key="item.id" v-bind:value="item.id">{{item.title}}</option>
+                </select>
+            </div>
+        </div>
+
+        <!-- TODO: Consider installing Vue Bootstrap and using a VB modal -->
+        <div class="modal fade" id="TitleModal" role="dialog">
+            <div class="modal-dialog">
+                <div class="modal-content">
+                    <div class="modal-header">
+                        <h4 class="modal-title">Add a new deck</h4>
+                        <button type="button" id="btnCloseTitleModal" class="close pull-right" data-dismiss="modal">&times;</button>
+                    </div>
+                    <div class="modal-body">
```

Comment:
I would put this modal into a component. 
This way we can call this component anywhere in the project to add a deck.
https://vuejs.org/v2/examples/modal.html

Reply:
I've put the modal into a component. It could use a little bit of visual work some as a transition however that is not important at the moment.

Reply:
There are a few queries I have though:

- Would it be better to isolate any logic which only affects the new deck component within the component itself - I.E. Call the API to add the deck in the new deck component? Otherwise, if we add the component to a different Vue, surely we'll essentially end up duplicating the API call? (DRY)

- Leading on from the above, the textbox is no longer cleared after the user adds a new deck. I am already using a reference to update some data which I feel is a little messy. To clear the textbox on success, I would have to call a function via in the component from the Vue via a reference, and then in that component, do the same thing again to update the data in this new component and hide the modal. This is of course incredibly messy.

If the API call were directly in the component, it would just be a case of updating the data on success.

Reply:
Both very good points.
there are multiple ways we can handle this and a quick google search have yielded the following  results:

- Using a repository factory (https://medium.com/canariasjs/vue-api-calls-in-a-smart-way-8d521812c322)
- or using vuex (https://appdividend.com/2018/05/08/vuex-axios-get-request-tutorial-with-example/)

now i would probably lean more towards the vuex way  as  it seems  to be more  standard  practice and is used in the following vuex boilerplate code.

https://github.com/chrisvfritz/vue-enterprise-boilerplate/tree/master/src/state/modules

I would recommend getting this PR sorted first and then once it is merged. Branch to work on one of the above methods.

Reply:
Fair enough. In that case, I will just add a transition to the modal and clear the textbox by using references for now. Once we have merged, we can look at refactoring using vuex if that sounds good to you?